jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/ResourceFactory.java

```diff
@@ -390,6 +439,62 @@ default Resource newJarFileResource(URI uri)
         return newResource(URIUtil.toJarFileUri(uri));
     }
 
+    /**
+     * Split a string of references, that may be split with '{@code ,}', or '{@code ;}', or '{@code |}' into URIs.
+     * <p>
+     *     Each part of the input string could be path references (unix or windows style), or string URI references.
+     * </p>
+     * <p>
+     *     If the result of processing the input segment is a java archive, then its resulting URI will be a mountable URI as {@code jar:file:...!/}
+     * </p>
+     *
+     * @param str the input string of references
+     */
+    default List<Resource> split(String str)
+    {
+        List<Resource> list = new ArrayList<>();
+
+        StringTokenizer tokenizer = new StringTokenizer(str, ",;|");
```

Comment:
This has traditionally been a static that works well, but in reality it has a few quirks.
Eg: The comma `,` and semicolon `;` are valid characters on all filesystems.

I believe the pipe `|` is also valid on a shocking large number of filesystems (even jar / war / zip too), but don't hold me to that.

Perhaps this needs some javadoc indicating that those characters cannot be used in filename? meh.